spring-integration-core/src/main/java/org/springframework/integration/graph/NodeInfo.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.graph;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.springframework.integration.support.context.NamedComponent;
+
+/**
+ * Base class for all endpoints.
+ *
+ * @author Gary Russell
+ * @since 4.3
+ *
+ */
+public abstract class NodeInfo {
+
+	private static final AtomicInteger nodeId = new AtomicInteger();
```

Comment:
Maybe this isn't good if we have several SI apps in the same JVM?
Can't we just provide an `id` from the `GraphBuilder` level from its context wide singleton perspective?